static/js/containers/DashboardPage.js

```diff
@@ -270,6 +314,51 @@ class DashboardPage extends React.Component {
     dispatch(setDocsInstructionsVisibility(bool));
   };
 
+  setCouponNotificationVisibility = bool => {
+    const { dispatch } = this.props;
+    dispatch(setCouponNotificationVisibility(bool));
+  };
+
+  setActiveCoupon = apiCoupon => {
+    const { dispatch } = this.props;
+    // convert `amount` from string to decimal
+    const coupon = R.evolve({amount: Decimal}, apiCoupon);
+    dispatch(setActiveCoupon(coupon));
+  };
+
+  renderCouponDialog() {
+    const {
+      programs,
+      ui,
+      coupons,
+    } = this.props;
+    if ( !coupons.activeCoupon ) {
+      return null;
+    }
+    let couponProgram = null, couponCourse = null;
+    if ( programs.availablePrograms ) {
+      couponProgram = programs.availablePrograms.find(
+        program => program.id === coupons.activeCoupon.program_id
+      );
+    }
+    if (
+      couponProgram &&
+      couponProgram.courses &&
```

Comment:
If `couponProgram` is of type `AvailableProgram`, there isn't a `courses` field. You'll need to use the dashboard API for that, or add it to the programs API, or change the dialog text to not use the course or run title